confidential-data-hub/hub/src/hub.rs

```diff
@@ -6,19 +6,85 @@
 use std::collections::HashMap;
 
 use async_trait::async_trait;
+use image_rs::image::ImageClient;
 use kms::{Annotations, ProviderSettings};
-use log::info;
+use log::{info, warn};
+use std::env;
 use storage::volume_type::Storage;
 
-use crate::{DataHub, Error, Result};
+use crate::{CdhConfig, DataHub, Error, Result};
 
 pub struct Hub {
     pub(crate) credentials: HashMap<String, String>,
+    image_client: tokio::sync::Mutex<image_rs::image::ImageClient>,
 }
 
 impl Hub {
-    pub async fn new(credentials: HashMap<String, String>) -> Result<Self> {
-        let mut hub = Self { credentials };
+    pub async fn new(config: CdhConfig) -> Result<Self> {
+        config.set_configuration_envs();
+        let credentials = config
+            .credentials
+            .iter()
+            .map(|it| (it.path.clone(), it.resource_uri.clone()))
+            .collect();
+
+        let mut image_client = match config.image.work_dir {
+            Some(work_dir) => ImageClient::new(work_dir.into()),
+            None => ImageClient::default(),
+        };
+
+        if let Some(uri) = config.image.authenticated_registry_credentials_uri {
+            if !uri.is_empty() {
+                image_client.config.auth = true;
+                image_client.config.file_paths.auth_file = uri;
+            }
+        }
+
+        if let Some(uri) = config.image.image_security_policy_uri {
+            if !uri.is_empty() {
+                image_client.config.security_validate = true;
+                image_client.config.file_paths.policy_path = uri;
+            }
+        }
+
+        if let Some(uri) = config.image.sigstore_config_uri {
+            if !uri.is_empty() {
+                image_client.config.file_paths.sigstore_config = uri;
+            }
+        }
+
+        if let Some(number) = config.image.max_concurrent_layer_downloads_per_image {
+            image_client.config.max_concurrent_download = number;
+        }
```

Comment:
Do you think it would make sense to move this logic into the initializer for the `image_client` (and pass `config.image` into that) rather than putting all this image-specific stuff in this function?

Reply:
There are some barriers because the parameters is different from `image-rs`'s. If you are worrying about the exposion of the args of `iamge_client`, we can add a `Builder` logic to `ImageClient`. If you are worrying about the duplication of settings, we could do a refactoring upon image-rs to have a `ClientBuilder` logic to make it better.

wdyt?